### PR TITLE
refactor(*): refactor imports, exports, and jsdocs

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: n/a (2018-08-10 free friday)

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### Before
* Requires importing `helix-ui` as a namespace to consume
* HelixUI exports can be verbose to consume
* `elements` namespace looks more like a property

```javascript
import HelixUI from 'helix-ui';

const ICON_NAMES = Object.keys(HelixUI.ICONS);

class MyElement extends HelixUI.elements.HXElement { ... }

HelixUI.initialize();
```

### After
* Importing `helix-ui` as a namespace is optional. You could import individual items.
* `elements` renamed to `Elements` to better communicate it as a namespace (**breaking change**)
* New `Utils` namespace

_Option 1: import module as namespace (retain status quo)_
```javascript
import * as HelixUI from 'helix-ui';

const ICON_NAMES = Object.keys(HelixUI.ICONS);
const KEY_MAP = Utils.KEYS;

class MyElement extends HelixUI.Elements.HXElement { ... }

HelixUI.initialize();
```

_Option 2: import module items separately_
```javascript
import { initialize, ICONS, Elements, Utils } from 'helix-ui';

initialize();

const ICON_NAMES = Object.keys(ICONS);
const KEY_MAP = Utils.KEYS;

class MyElement extends Elements.HXElement { ... }
```

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 017d5fa0e2bfd4e3e5a51189f15eb0f596b99a7f

https://app.netlify.com/sites/helix-ui/deploys/5b6e4391dd28ef378084a040

File: src/helix-ui/elements.js [hunk 1]
```diff
@@ -1,32 +0,0 @@
-/**
```
Comment (CITguy) [inline src/helix-ui/elements.js]: Moved to `src/helix-ui/elements/index.js`

File: src/helix-ui/elements/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+export { HXAccordionElement } from './HXAccordionElement';
```
Comment (CITguy) [inline src/helix-ui/elements/index.js]: * file moved from `src/helix-ui/elements.js`
* simplified import paths from `./elements/<HXFileName>.js` to `./<HXFileName>`

File: src/helix-ui/icons.js [hunk 3]
```diff
@@ -1,169 +0,0 @@
-import _account from './icons/account.svg';
```
Comment (CITguy) [inline src/helix-ui/icons.js]: File moved to `src/helix-ui/icons/index.js` with modified contents.

File: src/helix-ui/icons/index.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
+import _account from './account.svg';
```
Comment (CITguy) [inline src/helix-ui/icons/index.js]: simplified import paths from `./icons/<icon>.svg` to `./<icon>.svg`

File: src/helix-ui/index.js [hunk 5]
```diff
@@ -1,21 +1,22 @@
-/** @module helix-ui */
-import * as elements from './elements';
-import { version as SEMVER } from '../../package.json';
-import ICONS from './icons';
+/** @module HelixUI */
+import * as Elements from './elements';
+export { version as VERSION } from '../../package.json';
+export { default as ICONS } from './icons';
+export { default as Utils } from './utils';
```
Comment (CITguy) [inline src/helix-ui/index.js]: Adds `HelixUI.Utils.*` to generated JavaScript.

File: src/helix-ui/util.js [hunk 6]
```diff
@@ -1,3 +0,0 @@
-/** @module */
```
Comment (CITguy) [inline src/helix-ui/util.js]: Moved to `src/helix-ui/util/index.js`

**Comment by CITguy** [comment]
Marked "DO NOT MERGE". Will update to add deep module importing...

```javascript
import { ICONS } from 'helix-ui';
import { KEYS } from 'helix-ui/utils';
import { getPosition } from 'helix-ui/utils/position';
```

File: src/helix-ui/utils/index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    Shift: 16,
+    Space: 32,
+    Tab: 9,
+    Up: 38,
+};
+
+export default {
+    KEYS,
+    Position,
+};
```
Comment (cathysiller) [inline src/helix-ui/utils/index.js]: add the onScroll as well

File: src/helix-ui/utils/index.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+/**
+ * Communicate scroll events from elements at arbitrary depths in the DOM tree
+ * (because 'scroll' events do not bubble).
+ *
+ * The event is dispatched from the `document` object, instead of bubbling from
+ * the oritinal element, to avoid interfering with 'scroll' event listeners
```
Comment (cathysiller) [inline src/helix-ui/utils/index.js]: `original element`

**Comment by 100stacks** [comment]
Dev LGTM

**Pull request merged**